src/state/messages/convo/index.tsx

```diff
@@ -78,6 +82,23 @@ export function ConvoProvider({
     }, [convo, convoId, markAsRead]),
   )
 
+  React.useEffect(() => {
+    return convo.on(event => {
+      switch (event.type) {
+        case 'invalidate-block-state': {
+          for (const did of event.accountDids) {
+            queryClient.invalidateQueries({
+              queryKey: createProfileQueryKey(did),
+            })
+          }
+          queryClient.invalidateQueries({
+            queryKey: ListConvosQueryKey,
+          })
+        }
+      }
+    })
+  }, [convo, queryClient])
```

Comment:
Could we init `Convo` with `QueryClient` instead of the emitter for this?

Reply:
Since the life time of a queryClient should match the lifetime of a user, and therefore a `Convo`, yeah I think we could for now. But if in the future those lifetimes change, this would allow it to keep working.

Felt no-less stable to do it this way, so figured it's fine for now. What do you think?